crates/api_crud/src/site/read.rs

```diff
@@ -1,30 +1,32 @@
+use crate::user::my_user::get_my_user;
 use actix_web::web::{Data, Json};
-use lemmy_api_common::{
-  context::LemmyContext,
-  site::{GetSiteResponse, MyUserInfo},
-};
+use lemmy_api_common::{context::LemmyContext, site::GetSiteResponse};
 use lemmy_db_schema::source::{
-  actor_language::{LocalUserLanguage, SiteLanguage},
-  community_block::CommunityBlock,
-  instance_block::InstanceBlock,
+  actor_language::SiteLanguage,
   language::Language,
   local_site_url_blocklist::LocalSiteUrlBlocklist,
   oauth_provider::OAuthProvider,
-  person_block::PersonBlock,
   tagline::Tagline,
 };
 use lemmy_db_views::structs::{LocalUserView, SiteView};
-use lemmy_db_views_actor::structs::{CommunityFollowerView, CommunityModeratorView, PersonView};
-use lemmy_utils::{
-  build_cache,
-  error::{LemmyErrorExt, LemmyErrorType, LemmyResult},
-  CacheLock,
-  VERSION,
-};
+use lemmy_db_views_actor::structs::PersonView;
+use lemmy_utils::{build_cache, error::LemmyResult, CacheLock, VERSION};
 use std::sync::LazyLock;
 
 #[tracing::instrument(skip(context))]
-pub async fn get_site(
+pub async fn get_site_v3(
```

Comment:
These backward-compatibility hacks (and serving up api/v3 completely) should really be handled in client libraries IMO, not piling them on top of each other in the lemmy codebase. 

Not only is it going to get increasingly complicated, but we're going to be on the hook to support both of these alongside each other, with all the object changes.

IMO since this is a breaking changes release, we should just increment the api version, but don't support the old one.

Reply:
If we remove the v3 api directly in 0.20, it means that all clients and frontends will be completely broken until the devs rewrite their code for v4, which could take months. Even just dropping `GetSiteResponse.my_user` would probably break all authentication functionality, or even lead to crashes.

The backwards compatibility is really easy to implement in this case, and it prevents a lot of problems for users. Lemmy has a huge userbase now which relies on a certain level of stability, so even in major releases we cannot break everything just to save a little bit of maintenance work.

Reply:
I think dropping v3 support should wait for whenever we release Lemmy 1.0.0. A breaking change of that size corresponds well with what's communicated in the version number: a major version update that breaks backwards compatibility.